waspc/src/Wasp/Psl/Parser/Schema.hs

```diff
@@ -0,0 +1,37 @@
+module Wasp.Psl.Parser.Schema
+  ( parsePrismaSchema,
+    -- For testing only
+    schema,
+  )
+where
+
+import Text.Parsec
+  ( choice,
+    eof,
+    many,
+  )
+import qualified Text.Parsec as Parsec
+import Text.Parsec.String (Parser)
+import qualified Wasp.Psl.Ast.Schema as Psl.Schema
+import Wasp.Psl.Parser.Common (whiteSpace)
+import Wasp.Psl.Parser.ConfigBlock (configBlock)
+import Wasp.Psl.Parser.Enum (enum)
+import Wasp.Psl.Parser.Model (model)
+
+type SourceCode = String
+
+parsePrismaSchema :: SourceCode -> Either Parsec.ParseError Psl.Schema.Schema
+parsePrismaSchema = Parsec.parse schema ""
+
+schema :: Parser Psl.Schema.Schema
+schema = do
+  whiteSpace
+  elements <-
+    many $
+      choice
+        [ Psl.Schema.ModelBlock <$> model,
+          Psl.Schema.EnumBlock <$> enum,
+          Psl.Schema.ConfigBlock <$> configBlock
+        ]
+  eof
```

Comment:
Maybe put a comment here eplxaining why we put eof. I think the typical strategy for parsers is to let them work even if these is sometihng leftover. Here you forced eof, and that is ok, but is not so standard, so explanation might be useful.

Alternatively you could have applied that eof somewhere above, where we use this `schema` parser, so that `schema` parser remains more "open" and can be used in situations where we might not want to enforce eof, but then up there you can make that decision. If this was a library, that would certainly be the right choice, but in our case it doesn't matter that much, I doubt we will ever need to parse PSL whlie leaving sometihng after it.